### ISSUE TITLE
TypeError when dismissing Paywall ViewController

### ISSUE THREAD

**Issue by GaddiSunshine** [adaptyteam/AdaptyUI-React-Native#6]
When dismissing the paywall ViewController I get a TypeError.

This is a very general type error. So ignoring the warning in debug mode and on Sentry is not preferable. Not ignoring it will however explode Sentry Logs.

https://github.com/adaptyteam/AdaptyUI-React-Native/assets/55955275/183ed0b3-1d2f-434b-8171-a59e971327c8

Expected behaviour:
After dismissing the Paywall ViewController, I would not get a type error.

Actual behaviour:
After dismissing the Paywall ViewController, i get a type error.

Platform:
- iOS
- Android unconfirmed (unable to build debug android build with the current packages.)

Versions:
```
"react-native": "0.72.6",
"expo": "49.0.18",
"react-native-adapty": "2.7.0",
"@adapty/react-native-ui": "2.0.1",
```

Minimal reproducible example:
```
import React, { FC, useCallback, useEffect, useRef } from 'react'
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native'
import { createPaywallView } from '@adapty/react-native-ui'
import { ViewController } from '@adapty/react-native-ui/dist/view-controller'
import { adapty } from 'react-native-adapty'

const AdaptyPaywallBuilder: FC = () => {
  const paywallViewController = useRef<ViewController>()

  const initializePaywallViewController = useCallback(async () => {
    const paywall = await adapty.getPaywall('paywallbuilder-test')
    paywallViewController.current = await createPaywallView(paywall)
    paywallViewController.current.registerEventHandlers({
      onCloseButtonPress: () => true,
    })
  }, [])

  useEffect(() => {
    initializePaywallViewController()
  }, [initializePaywallViewController])

  const presentPaywall = useCallback(() => {
    if (paywallViewController.current) {
      paywallViewController.current.present()
    }
  }, [])

  return (
    <View style={styles.root}>
      <TouchableOpacity style={styles.button} onPress={presentPaywall}>
        <Text style={styles.text}>Present Paywall</Text>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    backgroundColor: 'green',
    padding: 16,
  },
  text: {
    fontSize: 14,
    fontWeight: '600',
  },
})

export default AdaptyPaywallBuilder
```

Logdump
```
TypeError: Cannot read property 'forEach' of undefined
TypeError: Cannot read property 'forEach' of undefined
    at removeAllEventListeners (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:329824:32)
    at unsubscribe (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:345525:48)
    at ?anon_0_ (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:345467:41)
    at next (native)
    at fulfilled (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:190966:32)
    at tryCallOne (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:33534:16)
    at anonymous (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:33615:27)
    at apply (native)
    at anonymous (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:34998:26)
    at _callTimer (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:34877:17)
    at _callReactNativeMicrotasksPass (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:34922:17)
    at callReactNativeMicrotasks (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:35128:44)
    at __callReactNativeMicrotasks (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:6564:46)
    at anonymous (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:6338:45)
    at __guard (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:6537:15)
    at flushedQueue (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:6337:21)
    at invokeCallbackAndReturnFlushedQueue (http://192.168.100.57:8081/index.bundle//&platform=ios&dev=true&hot=false&lazy=true:6331:33)
```

**Comment by vladd-g** [comment]
@GaddiSunshine thank you for the report! We've just released updates, could you please check it on the `2.9.3` and `2.1.0` respectively?

**Comment by GaddiSunshine** [comment]
> @GaddiSunshine thank you for the report! We've just released updates, could you please check it on the `2.9.3` and `2.1.0` respectively?

This has been resolved and no longer happens in versions `2.9.3` and `2.1.0`. Thank you for the quick update!
